### ISSUE TITLE
should getDistributedObjects include `__sql.catalog` map ?

### ISSUE THREAD

**Issue by srknzl** [hazelcast/hazelcast#19006]
**Describe the bug**
I think this is an internal map for SQL and should be hidden.
**Expected behavior**
I expect it to be filtered from getDistributedObjects results.

**To Reproduce**

1. Run 5.0 snapshot server(latest) `docker run -p 5701:5701 hazelcast/hazelcast:5.0-SNAPSHOT`
2. Run the following script

```java
import com.hazelcast.client.config.ClientConfig;
import com.hazelcast.client.HazelcastClient;
import com.hazelcast.core.HazelcastInstance;

public class Main {
    public static void main(String[] args) {
        ClientConfig c = new ClientConfig();
        HazelcastInstance h = HazelcastClient.newHazelcastClient(c);
        System.out.println(h.getDistributedObjects().size()); // 1
        System.out.println(h.getDistributedObjects().toArray()[0]); // ReplicatedMap{name='__sql.catalog'}
    }
}

```

**Comment by gurbuzali** [comment]
as a note, we have also maps with the prefix `__jet.`

**Comment by viliam-durina** [comment]
A second note, if we start hiding maps, then should the users also be able to query?
```java
instance.getMap("__sql.catalog").entrySet();
instance.getSql().execute("select * from \"__sql.catalog\"");
```

**Comment by viliam-durina** [comment]
I think this is a rabbit hole. I wouldn't hide the internal maps, at least not on this level. We won't be able to hide them completely from all public API, so why hide it just in `getDistributedObjects`? We can optionally hide them in MC and maybe and even there we should have an option `Hide internal IMaps` that would be on by default.

**Comment by srknzl** [comment]
I raised this to inform you about this behavior. I don't have a strong opinion. You(those that are responsible for this API) can decide what to do.

**Comment by Danny-Hazelcast** [comment]
hzInstance.getDistributedObjects()
returns some internal  map named  __sql.catalog   previously internal structs where not returned

**Comment by viliam-durina** [comment]
@Danny-Hazelcast because there were no internal maps in IMDG. There were such maps in Jet and they were returned.

I think this is a product decision (cc @Holmistr). My opinion is that if we cannot hide these objects in full (including preventing any access to them from clients), then we should not pretend they don't exist.

**Comment by mmedenjak** [comment]
A couple of more points to consider.

1. The contract of `getDistributedObjects` isn't really broken with the introduction of internal maps. We just didn't think about them before.
```
    /**
     * Returns all {@link DistributedObject}s, that is all maps, queues,
     * topics, locks etc.
     * <p>
     * The results are returned on a best-effort basis. The result might miss
     * just-created objects and contain just-deleted objects. An existing
     * object can also be missing from the list occasionally. One cluster
     * member is queried to obtain the list.
     *
     * @return the collection of all instances in the cluster
     */
    Collection<DistributedObject> getDistributedObjects();
```

2. If we start hiding the objects at the API level, we may break the MC behaviour where the objects were optionally hidden, depending on the switch.

**Comment by Holmistr** [comment]
Unless there's a way of completely preventing access to the maps for the user, we should stick to the contract of getDistributedObjects.
I don't see this as a critically confusing thing as it doesn't break anything for the users (some extra maps are returned, not a big deal). Let's handle this with documentation (CC @Serdaro) and I would also suggest adding this information about internal maps directly to the JavaDoc of getDstributedObjects. 
WDYT @mmedenjak @viliam-durina ?

**Comment by mmedenjak** [comment]
Yes, sure, and I see Viliam agrees as well.

**Comment by AyberkSorgun** [comment]
@Serdaro We can close this issue after documenting it properly

**Comment by Serdaro** [comment]
@AyberkSorgun , @mmedenjak , I can put a note under the example code shown at https://docs.hazelcast.com/hazelcast/latest/data-structures/distributed-data-structures.html#example-distributed-object-code saying that "the `getDistributedObjects()` method may return some extra distributed objects (maps, queues, etc.) which are internal to API, such as `__sql.catalog"; you can ignore them while working with your distributed objects"?

**Comment by mmedenjak** [comment]
Yes, something like that. Currently I think we have internal maps starting with `__sql.` and `__jet.` but I guess there might be more in the future.
